cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/kafkaassigner/KafkaAssignerDiskUsageDistributionGoal.java

```diff
@@ -463,8 +481,11 @@ Replica findReplicaToSwapWith(Replica replica,
   private boolean possibleToMove(Replica replica, Broker destinationBroker, ClusterModel clusterModel) {
     TopicPartition tp = replica.topicPartition();
 
-    boolean case1 = !clusterModel.partition(tp).partitionRacks().contains(destinationBroker.rack());
-    boolean case2 = replica.broker().rack() == destinationBroker.rack() && destinationBroker.replica(tp) == null;
+    boolean case1 = clusterModel.partition(tp).partitionRacks().stream()
+                                .map(this::mappedRackIdOf)
+                                .noneMatch(mappedRackIdOf(destinationBroker)::equals);
```

Comment:
I think it's making redundant calls to mappedRackIdOf(destinationBroker) for each stream element. If someone creates complicated logic in mappedRackIdOf, then it would impact the performance.

Reply:
fixed

Reply:
For this
```java
    .noneMatch(mappedRackIdOf(destinationBroker)::equals);
```

It is only one `mappedRackIdOf(destinationBroker)` call. `mappedRackIdOf(destinationBroker)` is evaluated to an String first, and then the method reference `evaluatedStr::equals` is passed in.